### PR TITLE
feat: enhancement of edit profile, follow-unfollow,comments feature

### PR DESCRIPTION
**Pull request by mariyasada**
-user can follow and unfollow another user
-user can only see the post of followed user
-user can edit profile
-user can comment on post and delete post

### CODE REVIEW

File: src/pages/userProfile/userProfile.jsx [hunk 1]
```diff
@@ -1,60 +1,102 @@
...
 export const UserProfile = () => {
   const { Posts } = useSelector((state) => state.post);
-  const { user } = useSelector((state) => state.auth);
+  const { user, userProfileData, updateDataStatus, getUserProfileStatus } =
+    useSelector((state) => state.auth);
   const [isShow, setShow] = useState(false);
+  const dispatch = useDispatch();
+
+  const { currentUserId } = useParams();
+  console.log(currentUserId);
```
Comment (tushartiwari7) [inline src/pages/userProfile/userProfile.jsx:16]: remove consoles

**Review by tushartiwari7 — APPROVED** [review]
Everything looks good
Nicely written
Good to merge.

File: src/components/CreatePost/CreatePost.jsx [hunk 2]
```diff
@@ -63,22 +59,24 @@ export const CreatePost = ({
...
-        <img
-          className="avatar avatar-xsm"
-          src={user.photoURL ? user.photoURL : "https://picsum.photos/200"}
-          alt="user profile"
-        />
+        <Link to={`/profile/${user.id}`}>
+          <img
+            className="avatar avatar-xsm"
+            src={user.photoURL ? user.photoURL : "https://picsum.photos/200"}
+            alt="user profile"
```
Comment (paarthbajaj) [inline src/components/CreatePost/CreatePost.jsx:66]: alt attribute could be made dynamic as well

**Review by paarthbajaj — APPROVED** [review]
Great piece of work. UI is good as well. Functionality is working super smooth. Appreciate it. Goodluck!

**Pull request merged**